pkg/registry/apps/daemonset/storage/storage.go

```diff
@@ -108,3 +137,67 @@ func (r *StatusREST) GetResetFields() map[fieldpath.APIVersion]*fieldpath.Set {
 func (r *StatusREST) ConvertToTable(ctx context.Context, object runtime.Object, tableOptions runtime.Object) (*metav1.Table, error) {
 	return r.store.ConvertToTable(ctx, object, tableOptions)
 }
+
+// ScaleREST implements a Scale for DaemonSet.
+type ScaleREST struct {
+	store *genericregistry.Store
+}
+
+// ScaleREST implements only Getter, /scale is read-only by design
+var _ = rest.Getter(&ScaleREST{})
+var _ = rest.GroupVersionKindProvider(&ScaleREST{})
+
+// GroupVersionKind returns GroupVersionKind for DaemonSet Scale object
+func (r *ScaleREST) GroupVersionKind(containingGV schema.GroupVersion) schema.GroupVersionKind {
+	switch containingGV {
+	case appsv1beta1.SchemeGroupVersion:
+		return appsv1beta1.SchemeGroupVersion.WithKind("Scale")
+	case appsv1beta2.SchemeGroupVersion:
+		return appsv1beta2.SchemeGroupVersion.WithKind("Scale")
+	default:
+		return autoscalingv1.SchemeGroupVersion.WithKind("Scale")
```

Comment:
There's no reason to support anything other than autoscaling/v1 Scale … this pattern was only for backwards compatibility with weird old Scale types on old scale subresources for existing types

Reply:
Thanks, removed it.